iothub/device/src/Authentication/IotHubConnectionInfo.cs

```diff
@@ -3,22 +3,26 @@
 
 using System;
 using System.Diagnostics;
+using System.Net;
 using System.Threading.Tasks;
 using Microsoft.Azure.Devices.Client.Extensions;
+using Microsoft.Azure.Devices.Client.Transport;
 
 namespace Microsoft.Azure.Devices.Client
 {
-    internal class IotHubConnectionInfo : IAuthorizationProvider
+    internal class IotHubConnectionInfo : IClientIdentity, IAuthorizationProvider
     {
-        private const int DefaultAmqpSecurePort = 5671;
-
-        public IotHubConnectionInfo(IotHubConnectionStringBuilder builder)
+        public IotHubConnectionInfo(
+            IotHubConnectionStringBuilder builder,
+            IotHubClientOptions iotHubClientOptions)
         {
             if (builder == null)
             {
                 throw new ArgumentNullException(nameof(builder));
             }
 
+            builder.Validate();
```

Comment:
We were previously used the following to create an IotHubConnectionInfo instance. Since this was called from an IotHubConnectionStringBuilder instance, this will not have the ClientOptions. The important bit in the method call below, however, is the call to Validate. I've moved it to the constructor.

```csharp
        internal IotHubConnectionInfo ToIotHubConnectionInfo()
        {
            Validate();
            return new IotHubConnectionInfo(this);
        }
```